plugins/@grouparoo/spec-helper/src/lib/factories/record.ts

```diff
@@ -1,14 +1,20 @@
-import { GrouparooRecord } from "@grouparoo/core";
+import { GrouparooModel, GrouparooRecord } from "@grouparoo/core";
 import faker from "faker";
 import { loadPath } from "../loadPath";
+import ModelFactory from "./model";
 
-const data = async (props = {}) => {
+const data = async (props: { modelId?: string } = {}) => {
   const { GrouparooModel } = await import(`@grouparoo/core/${loadPath}`);
+  const model =
+    (props.modelId
+      ? await GrouparooModel.findOne({ where: { id: props.modelId } })
+      : await GrouparooModel.findOne()) ??
+    ((await ModelFactory({ id: props.modelId })) as GrouparooModel);
```

Comment:
For the other single-instance factories which require a modelId, it's now optionally passed in.

Reply:
See my comment in `plugins/@grouparoo/spec-helper/src/lib/factories/destination.ts`